test/extended/baremetal/helper.go

```diff
@@ -0,0 +1,198 @@
+}
+
+// CreateExtraWorker creates a new BaremetalHost (and associated Secret).
+// If successfull, returns the newly created host and secret resources
+func (b *BaremetalTestHelper) CreateExtraWorker(host *unstructured.Unstructured, secret *v1.Secret) (*unstructured.Unstructured, *v1.Secret) {
+
+	secret, err := b.clientSet.CoreV1().Secrets("openshift-machine-api").Create(context.Background(), secret, metav1.CreateOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	host, err = b.bmcClient.Create(context.Background(), host, metav1.CreateOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	b.hostsCreated = append(b.hostsCreated, host)
+
+	return host, secret
+}
+
+// DeleteAllExtraWorkers deletes all the extra workers created in the current session
+func (b *BaremetalTestHelper) DeleteAllExtraWorkers() {
+	for _, host := range b.hostsCreated {
+		hostName := host.GetName()
+
+		err := b.bmcClient.Delete(context.Background(), hostName, metav1.DeleteOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		b.waitForDeletion(host)
+	}
+}
+
+func (b *BaremetalTestHelper) waitForDeletion(host *unstructured.Unstructured) {
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
+		_, err = b.bmcClient.Get(context.TODO(), host.GetName(), metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return true, nil
+			}
+
+			return false, err
+		}
+		return false, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// WaitForProvisioningState waits for the given baremetal host to reach the specified provisioning state.
+// If successfull, returns the updated host resource, otherwise the test will fail
+func (b *BaremetalTestHelper) WaitForProvisioningState(host *unstructured.Unstructured, expectedProvisioningState string) *unstructured.Unstructured {
+
+	hostName := host.GetName()
+	var newHost *unstructured.Unstructured
+
+	g.By(fmt.Sprintf("wait until %s becomes %s", hostName, expectedProvisioningState))
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+		newHost, err = b.bmcClient.Get(context.TODO(), hostName, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+
+			return false, err
+		}
+
+		actual, found, err := unstructured.NestedString(newHost.Object, "status", "provisioning", "state")
+		if err != nil {
+			return false, err
+		}
+		if found {
+			return expectedProvisioningState == actual, nil
+		}
+		return false, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return newHost
+}
+
+func (b *BaremetalTestHelper) Setup() {
+
+	//This code will ensure that any previous extra worker will be cleaned up
+	//This should make easier the migration of the serial metal ipi jobs
+	//for the adoption of the new way of deployng extra worker
+	hosts, err := b.bmcClient.List(context.Background(), metav1.ListOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	for _, host := range hosts.Items {
```

Comment:
Clean up is already done in here https://github.com/openshift/origin/pull/26681/files#diff-7e730e2c7566572f6d8e0ef47a5b3ca546de7f0d1bd9f0468ff6bed4064a037cR140. Moreover, it waits deletion is completed. 

Is there any possibility that is executed?

Reply:
This block is in reality independent, and just required to support the migration, given that there could be other tests using the APPLY_EXTRA_WORKERS feature. Once we'll get rid of it, we'll remove it